### ISSUE TITLE
Godot crashes after closing a LSP session

### ISSUE THREAD

**Issue by rafaeldelboni** [godotengine/godot#36472]
**Godot version:**
3.2 branch with this https://github.com/godotengine/godot/pull/35864

**OS/device including version:**
Arch Linux 5.5.4

**Issue description:**
When closing the LSP client connection Godot crashes.

**Steps to reproduce:**
Open Godot with LSP enabled, start a connection with LSP server and then close it.
Tried with coc.nvim and netcat.

**Minimal reproduction project:**
`nc localhost 6008`
Ctrl + c to close the netcat session, Godot will crash and close.

### Logs
```
Godot Engine v3.2.1.rc.custom_build.9faa6e649 - https://godotengine.org
OpenGL ES 3.0 Renderer: Mesa DRI Intel(R) HD Graphics 630 (Kaby Lake GT2) 
 
Editing project: /home/delboni/Workpaces/godot/coin-dash (::home::delboni::Workpaces::godot::coin-dash)
Godot Engine v3.2.1.rc.custom_build.9faa6e649 - https://godotengine.org
OpenGL ES 3.0 Renderer: Mesa DRI Intel(R) HD Graphics 630 (Kaby Lake GT2) 
 
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /usr/lib/libc.so.6(+0x3bd70) [0x7f2907894d70] (??:0)
[2] /usr/lib/libc.so.6(cfree+0x20) [0x7f29078e4460] (??:0)
[3] Memory::free_static(void*, bool) (/home/delboni/Sources/godot/core/os/memory.cpp:178)
[4] void memdelete<GDScriptLanguageProtocol::LSPeer>(GDScriptLanguageProtocol::LSPeer*) (/home/delboni/Sources/godot/./core/os/memory.h:119)
[5] Ref<GDScriptLanguageProtocol::LSPeer>::unref() (/home/delboni/Sources/godot/./core/reference.h:281)
[6] Ref<GDScriptLanguageProtocol::LSPeer>::~Ref() (/home/delboni/Sources/godot/./core/reference.h:296)
[7] GDScriptLanguageProtocol::poll() (/home/delboni/Sources/godot/modules/gdscript/language_server/gdscript_language_protocol.cpp:234)
[8] GDScriptLanguageServer::_notification(int) (/home/delboni/Sources/godot/modules/gdscript/language_server/gdscript_language_server.cpp:63)
[9] GDScriptLanguageServer::_notificationv(int, bool) (/home/delboni/Sources/godot/modules/gdscript/language_server/gdscript_language_server.h:39 (discriminator 14))
[10] Object::notification(int, bool) (/home/delboni/Sources/godot/core/object.cpp:933)
[11] SceneTree::_notify_group_pause(StringName const&, int) (/home/delboni/Sources/godot/scene/main/scene_tree.cpp:985)
[12] SceneTree::idle(float) (/home/delboni/Sources/godot/scene/main/scene_tree.cpp:524 (discriminator 3))
[13] Main::iteration() (/home/delboni/Sources/godot/main/main.cpp:2028)
[14] OS_X11::run() (/home/delboni/Sources/godot/platform/x11/os_x11.cpp:3243)
[15] /home/delboni/Sources/godot/bin/godot.x11.tools.64(main+0x121) [0x13f86a7] (/home/delboni/Sources/godot/platform/x11/godot_x11.cpp:57)
[16] /usr/lib/libc.so.6(__libc_start_main+0xf3) [0x7f2907880023] (??:0)
[17] /home/delboni/Sources/godot/bin/godot.x11.tools.64(_start+0x2e) [0x13f84ce] (??:?)
-- END OF BACKTRACE --
```

**Comment by akien-mga** [comment]
CC @ofrank123 @Geequlim

**Comment by ofrank123** [comment]
Was able to reproduce this, I'll work on a fix.

**Comment by ofrank123** [comment]
Should be fixed on the PR #36485.  Sorry about that.

**Comment by rafaeldelboni** [comment]
Just tested and is perfectly working now, thanks again for your amazing work.
I will keep testing this new LSP :)
